course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,101 @@
+#include "my_string.h"
+
+#include <cstring>
+#include <stdexcept>
+
+MyString::MyString(): mSize(0) { }
+
+MyString::MyString(const char *inchar) :mSize(std::strlen(inchar)), mBuffer(new char[mSize + 1])
+{
+	std::strcpy(mBuffer.get(), inchar);
+}
+
+MyString::MyString(const std::size_t size) :mSize(size), mBuffer(new char[mSize + 1])
```

Comment:
If you find this constructor useful n your code, maybe you could make it private, so it isn't part of the class' API